src/ripple/app/paths/impl/StrandFlow.h

```diff
@@ -181,7 +181,18 @@ flow (
         {
             EitherAmount stepIn (limitStepOut);
             for (auto i = limitingStep + 1; i < s; ++i)
-                stepIn = strand[i]->fwd (*sb, *afView, ofrsToRm, stepIn).second;
+            {
+                auto const r = strand[i]->fwd (*sb, *afView, ofrsToRm, stepIn);
+                if (strand[i]->dry (r.second) ||
+                    !strand[i]->equalIn (r.first, stepIn))
+                {
+                    // limiting step forward
```

Comment:
It might be nice to expand the comment here.  I think the point is that a `fwd` step should never limit because the `rev` pass should already have found all of the limits.